pyramid_openapi3/tween.py

```diff
@@ -0,0 +1,53 @@
+"""A tween to validate openapi responses."""
+from .exceptions import ResponseValidationError
+from .wrappers import PyramidOpenAPIRequest
+from .wrappers import PyramidOpenAPIResponse
+from pyramid.httpexceptions import HTTPNotFound
+from pyramid.request import Request
+from pyramid.response import Response
+from pyramid.tweens import reraise
+
+import sys
+import typing as t
+
+
+def response_tween_factory(handler, registry) -> t.Callable[[Request], Response]:
+    """Create response validation tween.
+
+    This tween should run after pyramid exception renderer view, so that
+    final response status and content_type are known and can be validated.
+
+    The only problem here is, that when response validation fails, we have
+    to return some exception response, with an unknown content type.
+    The advantage is, that these are server errors, and if 500 errors are
+    only possible due to response validation errors we don't need to document
+    them in the openapi spec file.
+    """
+
+    def excview_tween(request) -> Response:
+        try:
+            response = handler(request)
+            if not request.environ.get("pyramid_openapi3.validate_response"):
+                # not an openapi view or response validation not requested
+                return response
+            # validate response
+            open_request = PyramidOpenAPIRequest(request)
+            open_response = PyramidOpenAPIResponse(response)
+            settings = request.registry.settings["pyramid_openapi3"]
+            result = settings["response_validator"].validate(
+                request=open_request, response=open_response
+            )
+
+            if result.errors:
+                raise ResponseValidationError(errors=result.errors)
+        # If there is no exception view, we also see request validation errors here
+        except ResponseValidationError:
```

Comment:
I think a test is missing here for `if not request.environ.get('pyramid_openapi3.validate_response')` incase the handler raises one of these errors for some reason on a view that's not protected by the spec.

Reply:
Not quite sure whether I follow here, but I have added another test. https://github.com/Pylons/pyramid_openapi3/pull/26/commits/48b804907c5dfb430f77c8d936f043e6a5362437

Otherwise I think a non openapi view should not raise a `ResponseValidationError`. Maybe we should think about that in case there is a real use case for it, otherwise I'd consider throwing random `ResponseValidationError` instances in user code as a bug.